README.md

```diff
@@ -27,23 +27,45 @@ $ bower install particle-api-js
 
 ## Development
 
+
 All essential commands are available at the root via `npm run <script name>` - e.g. `npm run lint`. To view the available commands, run: `npm run`
 
 <details id="develop-run-tests">
 <summary><b>How to run your tests</b></summary>
 <p>
 
-to run the tests:
+Some of the tests depend on a real HTTP api backend and a valid access token. Be sure to set relevant environment variables to avoid test failures. You can prefix commands test commands like this `PARTICLE_API_BASE_URL=foo PARTICLE_API_TOKEN=bar npm test`
+
+`npm test` runs the tests.
+
+`npm run coverage` shows code coverage 
 
-`npm test`
+`npm run test:browser` runs browser tests with [karma](https://karma-runner.github.io/latest/index.html) (make sure you have the [Firefox launcher](https://npmjs.org/browse/keyword/karma-launcher) installed.
 
-to run the coverage:
+`npm run test:ci` runs tests in the exact same way CI system does
 
-`npm run coverage`
+</p>
+</details>
 
-to run browser tests with [karma](https://karma-runner.github.io/latest/index.html) (make sure you have the [Firefox launcher](https://npmjs.org/browse/keyword/karma-launcher) installed):
+<details id="develop-run-locally">
+<summary><b>How to write scripts that execute against local code changes?</b></summary>
+<p>
+
+Changes to source code happens in `src` directory. However, executable code that eventually ends up in a release gets compiled into `lib`.
+
+You can compile code via `npm run compile`.
+
+Node.js scripts that run against compiled code, typically start out like this:
+
+```js
+const Particle = require('./lib/Particle');
+const username = 'TBD';
+const password = 'TBD';
+const particle = new Particle();
+// Put specific api calls against `particle` here...
```

Comment:
can't make this a "suggestion" due to GitHub not handling the nested back-tics properly so i'll just include it below the line below:

- - - -

Source code lives in the `./src` directory and is built for release via the `npm run compile` command. To create a simple script file to test your changes, follow these steps:

1. create a `js` file on your local machine: `touch my-api-test.js`
2. within your test `js` file, init the api client like so:

```js
const ParticleAPI = require('./path/to/particle-api-js');
const api = new ParticleAPI();
```

3. add in any api calls, etc required to validate you changes

```js
const devices = await api.listDevices({ auth: '<particle-auth-token>' });
console.log('MY DEVICES:', devices);
```

4. run it: `node ./path/to/my-api-test.js`

_NOTE: Requiring the root directory works via the `main` field specified in Particle API JS' `package.json` file ([docs](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#main))

Reply:
Nice addition, improves clarity, added and tweaked slightly in https://github.com/particle-iot/particle-api-js/pull/143/commits/477729089f12e5a6d3a00055ab2ab68105fcef75